### ISSUE TITLE
Marten Queries against DateTimeOffset

### ISSUE THREAD

**Issue by albernhagen** [JasperFx/marten#2605]
Hello -- I am using Marten to query my entity which has a `DateTimeOffset` field.  With code like:

```
        var entities = session.Query<MyEntity>();
        var test = entities.Where(d => d.CreatedAt >= DateTimeOffset.UtcNow.AddDays(-7)).ToList();
```

I'm seeing a PostgresSQL query that look like:

`select d.data from public.mt_doc_myentity as d where public.mt_immutable_timestamptz(d.data ->> '') >= '6/7/2023 6:43:06 PM +00:00'`

It seems like the `->> ''` means it isn't mapping to the correct property as defined on my entity.  The entity field is simply:

```    
    [GraphQLDescription("The UTC timestamp when the entity was created")]
    [JsonInclude]
    [JsonProperty]
    public DateTimeOffset? CreatedAt { get; private set; }
```

Marten is registered as follows:

```
        services.AddMarten(options =>
        {
            // Establish the connection string to your Marten database
            options.Connection(connectionStringSettings.PostgresClientDb);

            options.Projections.Snapshot<MyEntity>(SnapshotLifecycle.Inline);

            options.AutoCreateSchemaObjects = AutoCreate.All;
            
            // Opt into System.Text.Json serialization
            options.UseDefaultSerialization(
                serializerType: SerializerType.SystemTextJson,
                // Optionally override the enum storage
                enumStorage: EnumStorage.AsString,
                // Optionally override the member casing
                casing: Casing.CamelCase
            );
        })
        .UseLightweightSessions();
```

This is on .NET 7 with Marten 6.0.1.  We recently upgraded to Marten 6.x and didn't have these issues before on Marten 5.x.

Thanks!

**Comment by albernhagen** [comment]
Upon posting this, I immediately found a workaround of adding the attribute of `[JsonPropertyName("createdAt")]` to my entity field.  Querying based on other field values works, so it seems like this is just needed with the `DateTimeOffset`,

**Comment by oskardudycz** [comment]
The issue comes from the empty `[JsonProperty]` attribute. It should be fixed by #2601 and released in v6.0.2.
